api/v1alpha1/client/affiliate.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+package sidero.discovery.client;
+
+option go_package = "github.com/talos-systems/api/v1alpha1/client/pb";
+
+// Affiliate protobuf spec, as seen internally in the client code.
+//
+// Discovery service recieves Affiliate marshaled to protobuf and encrypted.
+message Affiliate {
+    string node_id = 1;
+    repeated bytes addresses = 2;
+    string hostname = 3;
+    string nodename = 4;
+    string machine_type = 5;
+    string operating_system = 6;
+    optional KubeSpan kubespan = 7;
+}
+
+// KubeSpan optional configuration.
+message KubeSpan {
+    string public_key = 1;
+    bytes address = 2;
+    repeated IPPrefix additional_addresses = 3;
+}
+
+// IPPrefix contains CIDR.
+message IPPrefix {
+    bytes ip = 1;
+    int32 bits = 2;
+}
+
+// Endpoint for the Affiliate KubeSpan spec as seen internally in the client code.
+//
+// Discovery service receives Affiliate marshaled to protobuf and encrypted.
+message Endpoint {
+    bytes ip = 1;
```

Comment:
For IPv4 address, will that field contain 4 or 7-15 bytes? Maybe we could use `string` type to make that more clear?

If that's too late to change the type everywhere, we probably could add fields comments to explain that.

Reply:
it uses standard (more or less) binary marshaling for IPs which is implemented in `netaddr` library; I belive Linux kernel uses same at the very least, but good idea to have that documented

string representation is much larger

Reply:
Yeah, I'd rather stick with the binary form